compiler/src/main/kotlin/io/github/aplcornell/viaduct/backends/cleartext/CleartextCircuitCodeGenerator.kt

```diff
@@ -112,6 +121,128 @@ class CleartextCircuitCodeGenerator(context: CodeGeneratorContext) : AbstractCod
         }
     }
 
+    private fun createCommitment(
+        source: Protocol,
+        target: Protocol,
+        argument: Argument,
+        builder: CodeBlock.Builder,
+    ): CodeBlock {
+        require(context.host in source.hosts + target.hosts)
+        if (source !is Local) {
+            throw UnsupportedCommunicationException(source, target, argument.sourceLocation)
+        }
+        require(source.hosts.size == 1 && source.host in source.hosts)
+        require(target is CommitmentProtocol)
+        if (target.cleartextHost != source.host || target.cleartextHost in target.hashHosts) {
```

Comment:
You don't need to check `target.cleartextHost in target.hashHosts`, that's ensured by the `Commitment` class.